templates/_base-images-ref.html

```diff
@@ -0,0 +1,461 @@
+      })
+      .done(function (deviceTypes) {
+
+        var IGNORE_THESE_DEVICES = ['n310-tx2', 'apalis-imx6', 'colibri-imx6', 'cubox-i', 'edge', 'etcher-pro', 'edison', 'hummingboard2', 'odroid-u3+', 'odroid-ux3', 'nuc', 'parallella-hdmi-resin', 'solidrun-imx6']
+        var table = "";
+        var devices = {};
+
+        deviceTypes.d.forEach(function (device, idx) {
+          if (device.is_of__cpu_architecture[0].slug === "rpi") {
+            device.is_of__cpu_architecture[0].slug = "armv6hf";
+          }
+
+          if (!IGNORE_THESE_DEVICES.includes(device.slug)) {
+            if (device.is_of__cpu_architecture[0].slug in devices) {
+              devices[device.is_of__cpu_architecture[0].slug].push(device)
+            } else {
+              devices[device.is_of__cpu_architecture[0].slug] = [device]
+            }
+          }
+        }, "")
+
+        Object.keys(devices).forEach(function (key) {
+          table += "<h2>" + key + "</h2>"
+          devices[key].forEach(function (device) {
+              table += "<h3>" + device.name + "</h3>"
+              if (device.is_of__cpu_architecture[0].slug === "armv6hf") {
+                table += generateArmv6Table(device)
+              } else if (device.is_of__cpu_architecture[0].slug === "armv7hf") {
+                table += generateGenericTable(device)
+              } else if (device.is_of__cpu_architecture[0].slug === "aarch64") {
+                table += generateGenericTable(device)
+              } else if (device.is_of__cpu_architecture[0].slug === "amd64") {
+                table += generateGenericTable(device)
+              } else if (device.is_of__cpu_architecture[0].slug === "i386") {
+                table += generateI386Table(device)
+              } else {
+                table += "N/A"
+              }
+              
+            })
+        })
+
+        return window.jQuery("#base-images-ref").append(table);
+      })
+      .fail(function (error) {
+        return window
+          .jQuery("#base-images-ref")
+          .append(
+            '<h2>Something went wrong.</h2>'
+          );
+      });
+
+  }.call(this));
+
+  function generateGenericTable(deviceType) {
+    return '<table class="table table-bordered"> <thead> <tr> <th>Image</th> <th>Links</th> <th>Available Tag</th> </tr> </thead> <tbody>' +
+    
+    generateDebianTable(deviceType) +
+    generateAlpineTable(deviceType) +
+    generateFedoraTable(deviceType) +
+    generateUbuntuTable(deviceType) +
+
+    "</tbody> </table>";
+  }
+
+  function generateI386Table(deviceType) {
+    return '<table class="table table-bordered"> <thead> <tr> <th>Image</th> <th>Links</th> <th>Available Tag</th> </tr> </thead> <tbody>' +
+    
+    generateDebianTable(deviceType) +
+    generateAlpineTable(deviceType) +
+    generateUbuntuTable(deviceType) +
+
+    "</tbody> </table>";
+  }
+
+  function generateArmv6Table(deviceType) {
```

Comment:
How about changing `generateXYZTable` to use a generic `generateTable(deviceType, tableDefinitionOptions /* keyof typeof tableDefinitions */)` and abstract their implementation into the following definitions:
```js
var repoUrlsByLanguage = {
	python: $links.githubPythonBaseImages,
	golang: $links.githubGolangBaseImages,
	openjdk: $links.githubOpenJDKBaseImages,
	node: $links.githubNodeBaseImages,
	dontnet: $links.githubDotnetBaseImages,
};
var languages = [
	'python',
	'golang',
	'openjdk',
	'node',
];
var tableDefinitions = {
	alpine: {
		baseImagePrefixes: ['alpine']
		versions: '3.13 (latest), 3.11, 3.10, 3.12, edge',
		languages: languages,
	},
	armv6: {
		baseImagePrefixes : ['']
		versions: 'buster (latest), bullseye, jessie, stretch'
	},
	debian: {
		baseImagePrefixes : ['', 'debian']
		versions: 'buster (latest), bullseye, jessie, stretch'
		languages: languages.concat('dontnet'),
	},
	ubuntu: {
		baseImagePrefixes : ['ubuntu']
		versions: 'bionic (latest), cosmic, xenial, disco, eoan, focal'
		languages: languages,
	}
};
```
We can do that at a later point of course, but in that case lets open an issue referencing this comment.

Reply:
There are 2 kinds of generateXYZTable functions here, while `generateDebian|Ubuntu|Fedora|AlpineTable` are generic functions that generate contents for each distributions, the others will call the generic ones to correctly generate contents for it.

I think your suggestion will really help improve the current generic functions as we still have few things can be abstracted but it's better to move this to another PR after this one is merged

Reply:
Sorry for blocking this PR that long.
Yes, this can be a separate PR :+1: 
Let me open an issue so that we don't forget about this.